### PR TITLE
Upgrade StorageSync mgmt. SDK to 2022-09

### PR DESCRIPTION
**Pull request by jlindamood**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by jsquire** [comment]
@jlindamood:

Our automation is recognizing you as a community member.  Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct before moving this out of draft. 

- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 1]
```diff
@@ -177,16 +178,20 @@ public partial class StorageSyncRegisteredServerCollection : Azure.ResourceManag
...
+        public Azure.ResourceManager.StorageSync.Models.ServerAuthType? ActiveAuthType { get { throw null; } }
         public string AgentVersion { get { throw null; } set { } }
         public System.DateTimeOffset? AgentVersionExpireOn { get { throw null; } }
         public Azure.ResourceManager.StorageSync.Models.RegisteredServerAgentVersionStatus? AgentVersionStatus { get { throw null; } }
+        public string ApplicationId { get { throw null; } set { } }
         public System.Guid? ClusterId { get { throw null; } set { } }
         public string ClusterName { get { throw null; } set { } }
         public System.Uri DiscoveryEndpointUri { get { throw null; } set { } }
         public string FriendlyName { get { throw null; } set { } }
+        public bool? Identity { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:190]: Consider renaming this to `IsIdentity`

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 2]
```diff
@@ -1032,15 +1123,22 @@ public partial class StorageSyncRegisteredServerCreateOrUpdateContent : Azure.Re
     {
         public StorageSyncRegisteredServerCreateOrUpdateContent() { }
         public string AgentVersion { get { throw null; } set { } }
+        public string ApplicationId { get { throw null; } set { } }
         public System.Guid? ClusterId { get { throw null; } set { } }
         public string ClusterName { get { throw null; } set { } }
         public string FriendlyName { get { throw null; } set { } }
+        public bool? Identity { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:1130]: Consider renaming this to `IsIdentity`

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 3]
```diff
@@ -1032,15 +1123,22 @@ public partial class StorageSyncRegisteredServerCreateOrUpdateContent : Azure.Re
...
         public string LastHeartbeat { get { throw null; } set { } }
         public System.BinaryData ServerCertificate { get { throw null; } set { } }
         public System.Guid? ServerId { get { throw null; } set { } }
         public string ServerOSVersion { get { throw null; } set { } }
         public string ServerRole { get { throw null; } set { } }
     }
+    public partial class StorageSyncRegisteredServerPatch : Azure.ResourceManager.Models.ResourceData
+    {
+        public StorageSyncRegisteredServerPatch() { }
+        public bool? Identity { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:1140]: Consider renaming this to `IsIdentity`

File: sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md [hunk 4]
```diff
@@ -8,8 +8,8 @@ azure-arm: true
 csharp: true
 library-name: StorageSync
 namespace: Azure.ResourceManager.StorageSync
-require: https://github.com/Azure/azure-rest-api-specs/blob/d1eee5499dbf9281debdc90c4f4cbc7470fb8d6d/specification/storagesync/resource-manager/readme.md
-tag: package-2022-06-01
+require: https://github.com/Azure/azure-rest-api-specs/blob/9ec0fcc278aa2128c4fbb2b8a1aa93432d72cce0/specification/storagesync/resource-manager/readme.md
+tag: package-2022-09
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md:12]: Please specify the expected release date by updating the changelog.md and Azure.ResourceManager.StorageSync.csproj

**Comment by ArthurMa1978** [comment]
Close this pr after offline synced with @jlindamood as service release was delayed.

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Helpers/StorageSyncManagementTestUtilities.cs [hunk 5]
```diff
@@ -189,7 +189,8 @@ public static StorageSyncRegisteredServerCreateOrUpdateContent GetDefaultRegiste
...
...4QU1JSUJDZ0tDQVFFQXZPcHEzNWtnZVFGN2d2MnpwMGZ4dHdTTVFTMHlzRzFpdGpEaTFOOWVQUzZRDQo2QzJPcERjWjY3L1ZZTGdtSFJ0dzRuK3YxaG5PakJ1VVZMV0lwTGJHbW9TaGdRd2plNnBheExTYkNYWVJzYmJoDQpXK25hMDRIcWp0encxRnAwMERjNS8ybUFBclFLRWViRHRVbS9lbEdnZmh0UmpwS1RNMTkwUlcybnJxU3NRcWc2DQp4MWJYUEVHMWxoRmd1UUhEYVphMjMvUVZoSlFJalAzTjYwZWJWQ2ovOFdaeHFhVnlsU1ljaTJqa1BKUWVvUnY5DQpWWkhUNmpXWGRGRitZeFRlWi8vTHh0RkQwcEJsTEZMTCsrWmM2aHgvS0tObDVKU2NTVE81NmtFOGcrOFFPU3piDQpmaGp1MkVPNmVYUTdwUmdxb0wvb3h5K3dHSzUxUy9BRUtEMXdIRCtiZlFJREFRQUJvMUF3VGpBT0JnTlZIUThCDQpBZjhFQkFNQ0JhQXdIUVlEVlIwbEJCWXdGQVlJS3dZQkJRVUhBd0lHQ0NzR0FRVUZCd01CTUIwR0ExVWREZ1FXDQpCQlI5R2VpNXFrVHhiNFJBbGl6d0tZTEhJZVQxR0RBTkJna3Foa2lHOXcwQkFRc0ZBQU9DQVFFQW03RDhSOGdaDQpEZ2UveU5pbmtQd1pUY2xwRFhSVTVUcXlQbFR2Wld5L0dYV2k1SDVmbzlyWkh0ajYydlpBR1JLMHlrc2QvM0UrDQptVjFOTW42dldyaHB3bHpQMVBCTm1NTzhuTERveGQ4Q3ZTemZpVU91MDVOQUJEbVdHeWVURHgwWjRLbDlVTFVUDQpyWlNDS3U3TnYxYmpWOFYzRkNqdWFYaFhsL3ZTdFU2cGhUSzNYMEVlblJBaDRDRVF5NXpNYVFxSHd6cDArUXhXDQpCbDIvaHRkdWtNam1mVnpPeE5rSXQ1S2JWWmRjaVRPTFI2WTJPcUpSVlRMVE1KVE1rMHNESTZTRTI2b053Yk9NDQphSVRGWXlhVjczU3FGaXFmODFzOEVPYStLeHVLbkxWTWVIZWRZTTBGeXZFZ0UyTTF1UGg4cFZqUlJUcERmV1FvDQpldjJiUDJUYkRSU2U3UT09DQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tDQo="),
```
Comment (jlindamood) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Helpers/StorageSyncManagementTestUtilities.cs:193]: server cert

**Pull request closed without merging**